plugins/modules/net_tools/nmcli.py

```diff
@@ -905,6 +905,54 @@
                 description: C(NMSettingSecretFlags) indicating how to handle the I(wireguard.private-key) property.
                 type: int
                 choices: [ 0, 1, 2 ]
+    vpn:
+        description:
+            - The configuration to a VPN connection (PPTP and L2TP).
+            - In order to use L2TP you need to be sure that C(network-manager-l2tp) - and C(network-manager-l2tp-gnome) if host has UI - are installed on the host
+        type: dict
+        suboptions:
+            permissions:
+                description: User that will have persmission to use the connection
+                type: str
+                required: true
+            service-type:
+                description: This defines the service type of connection.
+                type: str
+                required: true
+                choices: [ pptp, l2tp ]
+            gateway:
+                description: The gateway to connection. It can be an IP (for example C(192.0.2.1)) or a FQDN address (for example C(vpn.example.com))
+                type: str
+                required: true
+            password-flags:
+                description:
+                    - NMSettingSecretFlags indicating how to handle the I(password) property.
+                    - 'Following choices are allowed:
+                      C(0) B(NONE): The system is responsible for providing and storing this secret (default),
+                      C(1) B(AGENT_OWNED): A user secret agent is responsible for providing and storing this secret; when it is required agents will be
+                           asked to retrieve it
+                      C(2) B(NOT_SAVED): This secret should not be saved, but should be requested from the user each time it is needed
+                      C(4) B(NOT_REQUIRED): In situations where it cannot be automatically determined that the secret is required
+                           (some VPNs and PPP providers do not require all secrets) this flag indicates that the specific secret is not required.'
+                type: int
+                choices: [ 0, 1, 2 , 4 ]
+                default: 0
+            user:
+                description: Username provided by VPN administrator
+                type: str
+                required: true
+            ipsec-enabled:
+                description:
+                    - Enable or disable IPSec tunnel to L2TP host.
+                    - This option is need when C(service-type) is C(l2tp)
+                type: bool
+                choices: [ yes, no ]
+            ipsec-psk:
+                description:
+                    - The Pre-shared key encoded.
+                    - You can encode using this linux command: C(echo "0s"$(base64 <<<'[YOUR PRE-SHARED KEY]' | rev | cut -c5- | rev))
```

Comment:
```suggestion
                    - >
                      You can encode using this linux command: C(echo "0s"$(base64 <<<'[YOUR PRE-SHARED KEY]' | rev | cut -c5- | rev))
```
Also why not provide an Ansible jinja2 expression on how to compute this?

I'm very curious where this magic number 5 comes from BTW. On the web I only found 2 so far for similar snippets, I guess the intention there was to get rid of the trailing `==`?

Reply:
I used an actual pre-shared key from an existing VPN connection on my workstation. I found the `cut -c2-` command too, but it didn't work. The magic 5 works in my lab. On the other hand, testing your jinja2 suggestion doesn't need the cut part.

```yaml
- ansible.builtin.set_fact:
    psk: "0s{{ ('Brittany123' | b64encode) }}"
```

If that's confusing, I'll cut it.

Reply:
That the magic 5 works might be due to something specific with the VPN connection you have. I would remove that from the documentation.